### DISCUSSION TITLE
How to show validation to custom blade columns?

### DISCUSSION THREAD

**Discussion by abhaypithadiya** [Power-Components/livewire-powergrid#1500] [Q&A] [answered]
I have a customised blade column, something like this:
```
            ->addColumn('product_remark', function ($productCollection) {
                return '<input wire:model="productRemark.'.$productCollection->id.'" type="text" class="form-control" placeholder="Enter Product Remark">';

            })
```

Validation Rules:
```
    protected function rules()
    {
        return [
            'productRemark' => ['required', 'array'],
            'productRemark.*' => ['required', 'max:255'],
        ];

    }
    
    $this->validate();
```

The validation works but I am not able to show the error message. I tried using ```@error``` but that doesn't work.

Thankyou!

**Comment by marineusde** [comment]
please show more code for us, how and where is $this->validate() called?

**Reply by abhaypithadiya** [reply]
I have an Action Column:
```
            Column::action('Action')
                ->visibleInExport(false),
```

```
    public function actions(ProductField $row): array
    {
        return [
            Button::add('edit')
                ->render(function (ProductField $row) {
                        return Blade::render(
                            <<<HTML
                        <a class="cursor-pointer" wire:click.prevent='addProduct($row)'></a>
                        HTML
                        );
                }),
        ];
    }
```
```
    public function addProduct($data)
    {
       $this->validate();
       
       // Other logic
    }
```

This is the flow. Thanks for the reply.

**Reply by marineusde** [reply]
the code in the first comment is for remark the product, the second is for adding the product. Sure thats the same? What do you want to validate, if the user just can add the product with a click?

**Reply by abhaypithadiya** [reply]
I am working on a project where the user can customise the type, shape and colour of the product, there are a lot of combinations for this. The client asked me to just give an input which holds the data of the same and that should be mandatory for every product they add. So Product Remark is required. The validation works but unable to show the message. I also tried using Blade::render.

**Reply by marineusde** [reply]
I would do that in a modal dialog, not in the table self. So your action button opens a livewire formular in a modal. there you can validate all the things the client want to be required. After saving the data in the form, you can close the modal and refresh the table.

**Reply by abhaypithadiya** [reply]
Sure, that is a work around but at the moment the client just wants the input field. If you have any solution, please let me know. Thanks!

**Comment by abhaypithadiya** [comment]
@dansysanalyst Any idea how I can achieve this?

**Reply by luanfreitasdev** [reply, marked as answer]
Hello! I tried a few things and I think I got a good result:

```php
->add('productRemark', function ($row) {
     $error = isset($this->getErrorBag()->getMessages()['productRemark.'.$dish->id])
        ? $this->getErrorBag()->getMessages()['productRemark.'.$dish->id][0]
        : null;

       return Blade::render('<x-editable :id="'.$row->id.'" errors="'.$error.'" />');
})
```

---

x-editable
```blade
@props([
    'id' => null,
    'errors' => null
])
<div>
    <input wire:model="productRemark.{{ $id }}" type="text" class="form-control"
        placeholder="Enter Product Remark">
    @if($errors)
        <div class="text-red-800">
            {!! $errors !!}
        </div>
    @endif
</div>
```

Result:
![image](https://github.com/Power-Components/livewire-powergrid/assets/33601626/2c6ade80-28f0-4f1e-9010-3f71a32a5128)

**Reply by abhaypithadiya** [reply]
@luanfreitasdev Hey! Sorry for the late reply. Thank you for the answer!
